sdk/go/common/resource/plugin/langruntime.go

```diff
@@ -16,22 +16,80 @@ package plugin
 
 import (
 	"context"
+	"fmt"
 	"io"
 
 	"github.com/blang/semver"
 	"github.com/hashicorp/hcl/v2"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	structpb "google.golang.org/protobuf/types/known/structpb"
 )
 
+// ProgramInfo contains minimal information about the program to be run.
+type ProgramInfo struct {
+	root       string
+	program    string
+	entryPoint string
+	options    map[string]any
+}
+
+func NewProgramInfo(rootDirectory, programDirectory, entryPoint string, options map[string]any) ProgramInfo {
+	return ProgramInfo{
```

Comment:
We should probably do some validation here now we've got the chance.
root and program directories should parse as valid paths, entryPoint should be a filename (i.e. no path separator). I think it's fine to panic if any of those don't validate as it'll be a system error not user input that got it wrong.

Reply:
Done ✅